### ISSUE TITLE
Show unencrypted content in `mkdocs serve`

### ISSUE THREAD

**Issue by joapuiib** [unverbuggt/mkdocs-encryptcontent-plugin#76]
I think a enhancement to editors using this plugin would be to be able to see the page content directly without entering a password while serving the site with `mkdocs serve`.

**Comment by unverbuggt** [comment]
You can set `remember_password: true`. This way the user only need to enter his credentials once in a `serve` session. Initially we had the option to save credentials to local storage, but local storage can be read from anyone who got access to the browser, which is not really secure.

Other way would be to make a copy of the "mkdocs.yml" and disable the plugin in there completely and then run `mkdocs serve -f test_mkdocs.yml`

**Comment by joapuiib** [comment]
Yeah. I've thought of `remember_password: true`, but I don't want to accidentally enable this feature on production.

At the moment, I am just commenting the password in the `page.meta` section. Again, there's a risk I forget to set it back.

I'd like to be able to see the content in development mode without changing the configuration.

I've thought of:

- Check if we are building/serving the site (`on_startup`) and save it.
- If we are serving, append the unencrypted content in the template.

I will implement this on my fork, but if you think this is useful to more users, maybe we can discuss how do you want to approach this.

**Comment by unverbuggt** [comment]
I think when a user does `mkdocs serve` he wants to have the same result as `mkdocs build` but with automatic rebuild on change.
That is not 100% accurate, as mkdocs injects some javascript to make the browser auto-reload on change, but nevertheless I wouldn't override the plugin.

If you use two different versions "mkdocs.yml", then you can easily separate development from production.

**Comment by joapuiib** [comment]
I get your point. But, if you're mantaining multiple projects, having different mkdocs would increase the efforts considerably.

In that sense, a option `show_on_serve` (or something similar) could be added (defaulting to False) that would allow rendering the page without a password.

**Comment by joapuiib** [comment]
I got a prototype going in https://github.com/joapuiib/mkdocs-encryptcontent-plugin/tree/feature/show-serving.

The changes are specified here: https://github.com/joapuiib/mkdocs-encryptcontent-plugin/commit/0b43fb2184c77ded72872d516cb19148f01a2d27

**Comment by unverbuggt** [comment]
I'd rather concentrate on slimming down the plugin of features most people won't need. It is as complex as it is because it grew in features over a long time.

Another idea would be to use [environment variables](https://www.mkdocs.org/user-guide/configuration/#environment-variables) to enable/disable the plugin ompletely if needed. We just need to add a switch in the plugin configuration.

**Comment by joapuiib** [comment]
Do you want me to try to implement this toggle?

**Comment by unverbuggt** [comment]
I found an [issue](https://github.com/mkdocstrings/mkdocstrings/issues/478) on the mkdocstrings plugin where someone had a similar suggestion. It lead to some [changes](https://github.com/mkdocstrings/mkdocstrings/pull/504) which were also [discussed](https://github.com/pawamoy/markdown-exec/issues/19) in the markdown-exec plugin. And there I found the [information](https://www.mkdocs.org/user-guide/configuration/#enabled-option) that this functionality is already implemented since mkdocs 1.6.
